File: src/test/java/AnotherServiceTest.java
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.when;

@ExtendWith(MockitoExtension.class)
public class AnotherServiceTest {
    @InjectMocks
    private AnotherService service;

    @Mock
    private SomeService someService;

    @Test
    public void testThrowsOnNonCalculateParameter() {
        final var exception = assertThrows(DomainException.class, () -> service.bar(false));
        assertEquals("Unexpected stuff happened", exception.getMessage());
    }

    @Test
    public void testSomeServiceResultUsed() {
        when(someService.foo()).thenReturn(42);
        final var result = assertDoesNotThrow(() -> service.bar(true));
        assertEquals("Some service returned 42", result);
    }
}

File: src/test/java/AnotherServiceTest.kt
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.junit.jupiter.MockitoExtension
import org.mockito.InjectMocks
import org.mockito.Mock
import org.junit.jupiter.api.Test
import kotlin.Throws
import org.mockito.Mockito.*

@ExtendWith(MockitoExtension::class)
class AnotherServiceTest {
    @InjectMocks
    private lateinit var service: AnotherService

    @Mock
    private lateinit var someService: SomeService

    @Test
    fun testThrowsOnNonCalculateParameter() {
        val exception = assertThrows(DomainException::class.java) { service.bar(false) }
        assertEquals("Unexpected stuff happened", exception.message)
    }

    @Test
    @Throws(DomainException::class)
    fun testSomeServiceResultUsed() {
        `when`(someService.foo()).thenReturn(42)
        val result = service.bar(true)
        assertEquals("Some service returned 42", result)
    }
}